src/main/java/com/redhat/red/build/finder/BuildFinder.java

```diff
@@ -519,8 +525,16 @@ public static void main(String[] args) {
             }
 
             LOGGER.debug("Configuration {} ", config);
-            // TODO: Decide if we should write out the config file or throw exception for missing file.
-            // JSONUtils.dumpFile(path.toFile(), config);
+
+            if (!configFile.exists()) {
+                File configDir = configPath.toFile().getParentFile();
+
+                if (configDir != null && !configDir.exists()) {
+                    configDir.mkdirs();
+                }
+
+                JSONUtils.dumpFile(configPath.toFile(), config);
+            }
```

Comment:
Should we write it out by default? That seems a bit unexpected?

Reply:
Based on your comment that was there, I thought you were OK with either. I suppose there is currently an issue when adding new config options, particularly with the default case.

Reply:
@dwalluck It meant there was was something to be discussed :-) Given that to function correctly it needs a brew URL I'm still not sure writing out, what could effectively be an invalid file, is going to be helpful?

Reply:
I think if it writes out a file, it's writing the current state, so it's neither better nor worse. If you specify the correct URLs on the command line, then it would write those options to the file instead.

There are a couple things we can do. We should be checking if the URLs match the default value and throwing an error before it even gets to the point of writing the config. Also, instead of writing the config unconditionally, I could add it as a command-line option. But, it's pretty common for apps to write out their config files without requiring user intervention.